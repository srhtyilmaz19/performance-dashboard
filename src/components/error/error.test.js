import React from "react";

import {cleanup, render} from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Error from "./error";

afterEach(cleanup)

describe('<Error />', () => {

    it('should render error component', function () {
        const {asFragment} = render(<Error/>)

        expect(asFragment()).toMatchSnapshot();
    });

    it('should have class', function () {
        const handleClick = () => console.log('handleClick triggered !');

        const {getByTestId} = render(<Error onClick={handleClick}/>)

        const button = getByTestId('error-component');

        expect(button).toHaveClass('loading-wrapper');
    });

    it('should text class', function () {
        const handleClick = () => console.log('handleClick triggered !');

        const {getByTestId} = render(<Error onClick={handleClick}/>)

        const button = getByTestId('error-component');

        expect(button).toHaveTextContent('Error. please try again later !');
    });

})
