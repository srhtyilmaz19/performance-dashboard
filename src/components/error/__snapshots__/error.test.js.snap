// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoData /> should render error component 1`] = `
<DocumentFragment>
  <div
    class="loading-wrapper"
    data-testid="error-component"
  >
    <p
      data-testid="error-text"
    >
      Error. please try again later !
    </p>
  </div>
</DocumentFragment>
`;
